File: src/main/kotlin/com/nazarethlabs/codex/service/note/FavoriteNoteService.kt
package com.nazarethlabs.codex.service.note

import com.nazarethlabs.codex.dto.Note
import com.nazarethlabs.codex.repository.NoteStorageRepository

class FavoriteNoteService {
    fun toggleFavorite(note: Note) {
        NoteStorageRepository
            .getInstance()
            .toggleFavorite(note.id)
    }
}

File: src/core/services/FavoriteNoteService.ts
import { Note } from '../dtos/Note';
import { NoteRepository } from '../repositories/NoteRepository';
import { DateHelper } from '../helpers/DateHelper';

export class FavoriteNoteService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  toggleFavorite(note: Note): void {
    const updatedNote: Note = {
      ...note,
      isFavorite: !note.isFavorite,
      updatedAt: DateHelper.nowMs(),
    };
    this.repository.save(updatedNote);
  }
}
